### ISSUE TITLE
Postgres POSITION() unparsable when including a JSON operator

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#6387]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I ran into "unparsable section" when linting code containing the `position()` function. It works fine for simple queries such as `SELECT position('#' IN 'abcd#e');` and `SELECT position('#' IN tbl.col_a) FROM tbl`, but arguments containing JSON operators such as `->` or `->>` result in "unparsable section".

### Expected Behaviour

Expected no parsing errors or linting failures.

### Observed Behaviour

```
== [stdin] FAIL                                                                                                                                                
L:   1 | P:  16 |  PRS | Line 1, Position 16: Found unparsable section: "('#' IN
                       | tbl.col_a -> 'b')"
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```bash
echo "SELECT position('#' IN tbl.col_a -> 'b') FROM tbl" | sqlfluff lint --dialect=postgres -
```

### Dialect

postgres

### Version

sqlfluff, version 3.2.0
Python 3.11.10

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
